### DISCUSSION TITLE
FCU A320 Push/Pull as a toggle for SPD, HDG, ALT, V/S instead of assign two buttons FSUIPC7

### DISCUSSION THREAD

**Discussion by magioni** [flybywiresim/aircraft#5847] [Help]
Is that possible to add simple "toggle" command to use only one button when mapping FSUIPC7 FCU fuctions such as :
Autopilot H on HDG:
click on button "x" mapped on FSUIPC7 = "managed" -> FCU controls of heading
another click same button "x" mapped on FSUIPC7 = "manual" FCU control of heading

This happens on Aerosoft A320 , instead of push and pull you can use same button for functions :
FCU A320 Push/Pull as a toggle for SPD, HDG, ALT, V/S instead of assign two buttons on FSUIPC7:  

Who does a button box use to have encoders only with push (not pull),

**Comment by MStallen** [comment]
If you have a box only with push-buttons like I have (VrInsight Combo) you can assign FC_HDG_PUSH to FSUIPC button pressed and FC_HDG_PULL to FSUIPC button released

**Comment by magioni** [comment]
Hi Mstallen, thank you I will check it  but I understand your point.

Actually I use a button box with 4 rotaries  (with push "on '' button
middle ) and other buttons.  I did something very similar to the FCU A320
btw . I use for P3D A320 Aerosoft FSLabs

My  rotaries encorder are only push keys in the middle.  They are not
"on/off" to be used as "when pressed" and "when released"" on FSUIPC.

AEROSOFT A320 for instance they have one assignment "toggle" , Push once
'manual FCU mode", push again "managed FCU mode", ...
That is what I´m trying
Than kyou




Em seg., 20 de set. de 2021 às 05:16, MStallen ***@***.***>
escreveu:

**Comment by MStallen** [comment]
You can also try something like this 👍 

In the FSUIPC.INI
[LvarOffsets]
0=A32NX_FCU_ALT_MANAGED=UB66C1

Then make a LUA named FCU_ALT.lua with
FCU_ALT = ipc.readUB("66C1")
if FCU_ALT == 0 then
        ipc.control(xxxxx)
else
        ipc.control(yyyyy)
end 

where xxxxx = controlnumber of A320_Neo_FCU_ALT_PUSH
and
where yyyyy = controlnumber of A320_Neo_FCU_ALT_PULL

Then assign Lua FCU_ALT to your button

**Comment by magioni** [comment]
Thanks for the tip!  I will che

Em qui., 23 de set. de 2021 às 06:25, MStallen ***@***.***>
escreveu:

**Comment by magioni** [comment]
Thanks, I'll check it out.

Em qui., 23 de set. de 2021 às 06:25, MStallen ***@***.***>
escreveu:

**Comment by magioni** [comment]
Thank you ! Actually I´m trying to understanding Lua and LVars programming
better, usually Linda would do the job :)
One question  where did you get info  *"A32NX_FCU_ALT_MANAGED=UB66C1*" ?



Em qui., 23 de set. de 2021 às 06:25, MStallen ***@***.***>
escreveu:

**Comment by magioni** [comment]
Thank you! I was able to make LINDA 4.0.5 work . So I´m able to push/pull
same button, as well as other functions


Em qui., 23 de set. de 2021 às 06:25, MStallen ***@***.***>
escreveu:
